TestList flakiness

#### Link(s) to logs

https://console.cloud.google.com/cloud-build/builds/746bede5-17c9-469a-9489-104094eadf51?project=ci-account

#### Relevant snippet

```
FAIL: github.com/gravitational/teleport/tool/tsh
FAIL: github.com/gravitational/teleport/tool/tsh.TestList
===================================================
OUTPUT github.com/gravitational/teleport/tool/tsh.TestList
===================================================
=== RUN   TestList
=== PAUSE TestList
=== CONT  TestList
=== CONT  TestList
    tsh_helper_test.go:250: 
        	Error Trace:	/workspace/tool/tsh/tsh_helper_test.go:250
        	            				/workspace/tool/tsh/proxy_test.go:627
        	Error:      	Condition never satisfied
        	Test:       	TestList
--- FAIL: TestList (19.92s)
```

--- a/tool/tsh/proxy_test.go
+++ b/tool/tsh/proxy_test.go
@@ -616,8 +616,6 @@ func TestEnvVarCommand(t *testing.T) {
 
 // TestList verifies "tsh ls" functionality
 func TestList(t *testing.T) {
-	t.Parallel()
-
 	isInsecure := lib.IsInsecureDevMode()
 	lib.SetInsecureDevMode(true)
 	t.Cleanup(func() {
@@ -630,6 +628,9 @@ func TestList(t *testing.T) {
 			cfg.Auth.NetworkingConfig.SetProxyListenerMode(types.ProxyListenerMode_Multiplex)
 		}),
 		withLeafCluster(),
+		withLeafConfigFunc(func(cfg *service.Config) {
+			cfg.Version = defaults.TeleportConfigVersionV2
+		}),
 	)
 	rootNodeAddress, err := s.root.NodeSSHAddr()
 	require.NoError(t, err)

